wrapper/src/main/java/software/amazon/jdbc/plugin/cache/CacheConnection.java

```diff
@@ -61,7 +84,39 @@ public class CacheConnection {
           "true",
           "Whether to use SSL for cache connections.");
 
+  protected static final AwsWrapperProperty CACHE_IAM_REGION =
+      new AwsWrapperProperty(
+          "cacheIamRegion",
+          null,
+          "AWS region for ElastiCache IAM authentication.");
+
+  protected static final AwsWrapperProperty CACHE_USERNAME =
+      new AwsWrapperProperty(
+          "cacheUsername",
+          null,
+          "Username for ElastiCache regular authentication.");
+
+  protected static final AwsWrapperProperty CACHE_PASSWORD =
+      new AwsWrapperProperty(
+          "cachePassword",
+          null,
+          "Password for ElastiCache regular authentication.");
+
+  protected static final AwsWrapperProperty CACHE_NAME =
+      new AwsWrapperProperty(
+          "cacheName",
+          null,
+          "Explicit cache name for ElastiCache IAM authentication. ");
+
   private final boolean useSSL;
+  private final boolean iamAuthEnabled;
+  private final String cacheIamRegion;
+  private final String cacheUsername;
+  private final String cacheName;
+  private final String cachePassword;
+  private final String awsProfile;
+  private final Properties awsProfileProperties;
```

Comment:
Those 2 fields are redundant. I would remove `awsProfile` if you are storing it inside `awsProfileProperties`.